C++/C.sublime-syntax

```diff
@@ -838,25 +841,28 @@ contexts:
               pop: true
             - include: preprocessor-disabled
 
+  preprocessor-if-true:
+    - match: ^\s*(#\s*endif)\b
+      captures:
+        1: meta.preprocessor.c keyword.control.import.c
+      pop: true
+    - match: ^\s*(#\s*else)\b
+      captures:
+        1: meta.preprocessor.c keyword.control.import.else.c
+      push:
+        - meta_content_scope: comment.block.preprocessor.else-branch.c
+        - match: (?=^\s*#\s*endif\b)
+          pop: true
+        - include: preprocessor-disabled
+
   preprocessor-rule-enabled-data-structures:
     - match: ^\s*((#if)\s+(0*1))\b
       captures:
         1: meta.preprocessor.c
         2: keyword.control.import.c
         3: meta.number.integer.decimal.c constant.numeric.value.c
       push:
-        - match: ^\s*(#\s*endif)\b
-          captures:
-            1: meta.preprocessor.c keyword.control.import.c
-          pop: true
-        - match: ^\s*(#\s*else)\b
-          captures:
-            1: meta.preprocessor.c keyword.control.import.else.c
-          push:
-            - meta_content_scope: comment.block.preprocessor.else-branch.c
-            - match: (?=^\s*#\s*endif\b)
-              pop: true
-            - include: preprocessor-disabled
+        - include: preprocessor-if-true
```

Comment:
In such situations a direct `push: preprocessor-if-true` would be preferable, because it would also apply any `meta_include_prototype`, `meta_scope` or `meta_content_scope` directive from the pushed context.